baseclient.go

```diff
@@ -96,6 +96,8 @@ func (client *BaseClient) ExecuteRequest(request *http.Request, body interface{}
 	}
 
 	data, err := ioutil.ReadAll(response.Body)
+	defer resp.Body.Close()
```

Comment:
Thanks, that's a good catch!!
We will soon be adding connection pooling in this SDK and this would've been a bummer if not caught